src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -457,6 +457,19 @@ IsDistributedTableRTE(Node *node)
 }
 
 
+/*
+ * IsReferenceTableRTE gets a node and returns true if the node
+ * is a range table relation entry that points to a reference table.
+ */
+bool
+IsReferenceTableRTE(Node *node)
+{
+	Oid relationId = NodeTryGetRteRelid(node);
+	return relationId != InvalidOid && IsCitusTable(relationId) &&
+		   PartitionMethod(relationId) == DISTRIBUTE_BY_NONE;
```

Comment:
Could use IsReferenceTable

Reply:
`IsReferenceTable` calls `GetCitusTableCacheEntry` which raises an error when `relationId` is not a citus table, making it rather misleading that the code checks `tableEntry->isCitusTable` when it'd've thrown if that weren't true

Reply:
PartitionMethod also calls GetCitusTableCacheEntry, so not sure that makes a tangible difference.